Automattic/jetpack
# Social: Add nextdoor embed block

This changes adds a new block to Jetpack that renders a Nextdoor post.

## Proposed changes:

* Added the JS for the block in the editor
* Added controls to change the URL for a present block
* Added rendering the iframe in PHP based on the attribute

### Other information:


## Jetpack product discussion


https://github.com/orgs/Automattic/projects/733/views/2?filterQuery=&pane=issue&itemId=41283324
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






* To see beta blocks add `define( 'JETPACK_BLOCKS_VARIATION', 'beta' );` to your `wp-config.php` file
* Create a new post and test the new Nextdoor blog
  * Typing `/nextdoor` should bring up the empty block
  * Insert a Nextdoor post url ( `https://nextdoor.com/p/<post_id>` ) or embed URL ( `https://nextdoor.com/embed/<post_id>` ), then click embed to turn it into an iframe.
  * Nextdoor has different domains ( `https://nextdoor.co.uk` ) those should work too
  * Using any other string should give an error
  * If you already embedded something into the block, you can change it in the block settings in the editor by clicking on the block
  * Pasting in any of the links above should instantly convert into the block
* Publish the post and test the following
  * Nextdoor blocks with valid URL should be rendered on the post
  * Invalid blocks should not get rendered on the post

| Editor | Published Post |
| ------------- | ------------- |
| ![CleanShot 2023-10-25 at 15 23 56 png](https://github.com/Automattic/jetpack/assets/36671565/7cd60cb9-595d-44ef-8370-5c25eb3e8464) | ![CleanShot 2023-10-25 at 15 24 20 png](https://github.com/Automattic/jetpack/assets/36671565/e32aad19-db81-4af5-bc3c-80366252687c) |
